tools/buildmgr/cbuildgen/config/TIARMCLANG.4.0.1.cmake

```diff
@@ -0,0 +1,217 @@
+# This file maps the CMSIS project options to toolchain settings.
+#
+#   - Applies to toolchain: Texas Instruments CGT Clang (TI-CGT-CLANG) Bare Metal Toolchain for the Arm Architecture 4.0.1 and greater
+
+set(AS "tiarmclang")
+set(CC "tiarmclang")
+set(CXX "tiarmclang")
+set(CPP "tiarmclang")
+set(OC "tiarmobjcopy")
+
+set(TOOLCHAIN_ROOT "${REGISTERED_TOOLCHAIN_ROOT}")
+set(TOOLCHAIN_VERSION "${REGISTERED_TOOLCHAIN_VERSION}")
+
+if(DEFINED TOOLCHAIN_ROOT)
+  set(PREFIX)
+  set(EXT)
+  
+  set(AS ${TOOLCHAIN_ROOT}/${PREFIX}${AS}${EXT})
+  set(CC ${TOOLCHAIN_ROOT}/${PREFIX}${CC}${EXT})
+  set(CXX ${TOOLCHAIN_ROOT}/${PREFIX}${CXX}${EXT})
+  set(CPP ${TOOLCHAIN_ROOT}/${PREFIX}${CPP}${EXT})
+  set(OC ${TOOLCHAIN_ROOT}/${PREFIX}${OC}${EXT})
+endif()
+
+# Helpers
+
+function(cbuild_set_defines lang defines)
+  set(TMP_DEFINES)
+  foreach(DEFINE ${${defines}})
+    string(REPLACE "\"" "\\\"" ENTRY ${DEFINE})
+    string(REGEX REPLACE "=.*" "" KEY ${ENTRY})
+    if (KEY STREQUAL ENTRY)
+      set(VALUE "1")
+    else()
+      string(REGEX REPLACE ".*=" "" VALUE ${ENTRY})
+    endif()
+    string(APPEND TMP_DEFINES "-D${ENTRY} ")
+  endforeach()
+  set(${defines} ${TMP_DEFINES} PARENT_SCOPE)
+endfunction()
+
+set(OPTIMIZE_VALUES       "debug" "none" "balanced" "size" "speed")
+set(OPTIMIZE_CC_FLAGS     "-Og"   "-O0"  "-O2"      "-Os"  "-O3")
+set(OPTIMIZE_ASM_FLAGS    ${OPTIMIZE_CC_FLAGS})
+set(OPTIMIZE_CXX_FLAGS    ${OPTIMIZE_CC_FLAGS})
+set(OPTIMIZE_LD_FLAGS     ${OPTIMIZE_CC_FLAGS})
+
+set(DEBUG_VALUES          "on"      "off")
+set(DEBUG_CC_FLAGS        "-g3"     "-g0")
+set(DEBUG_CXX_FLAGS       ${DEBUG_CC_FLAGS})
+set(DEBUG_LD_FLAGS        ${DEBUG_CC_FLAGS})
+set(DEBUG_ASM_FLAGS       ${DEBUG_CC_FLAGS})
+
+set(WARNINGS_VALUES       "on" "off" "all")
+set(WARNINGS_CC_FLAGS     ""   "-w"  "-Wall")
+set(WARNINGS_ASM_FLAGS    ""   "-w"  "-Wall")
+set(WARNINGS_CXX_FLAGS    ""   "-w"  "-Wall")
+set(WARNINGS_LD_FLAGS     ""   "-w"  "-Wall")
+
+set(LANGUAGE_VALUES       "c90"      "gnu90"      "c99"      "gnu99"      "c11"      "gnu11"      "c17"      "gnu17"      "c++98"      "gnu++98"      "c++03"      "gnu++03"      "c++11"      "gnu++11"      "c++14"      "gnu++14"      "c++17"      "gnu++17")
+set(LANGUAGE_CC_FLAGS     "-std=c90" "-std=gnu90" "-std=c99" "-std=gnu99" "-std=c11" "-std=gnu11" "-std=c17" "-std=gnu17" ""           ""             ""           ""             ""           ""             ""           ""             ""           "")
+set(LANGUAGE_CXX_FLAGS    ""         ""           ""         ""           ""         ""           ""         ""           "-std=c++98" "-std=gnu++98" "-std=c++03" "-std=gnu++03" "-std=c++11" "-std=gnu++11" "-std=c++14" "-std=gnu++14" "-std=c++17" "-std=gnu++17")
```

Comment:
Please add all language standard keys as described in the [schema](https://github.com/Open-CMSIS-Pack/devtools/blob/schemas/projmgr/2.6.0/tools/projmgr/schemas/common.schema.json#L183-L189).
It seems the following keys are missing: `"c23", "gnu23", "c++20", "gnu++20", "c++23", "gnu++23"`

Reply:
These standards are not supported by the TI toolchain.

Reply:
Missing keys will break the build.
I would recommend to set the non supported values to their best match or as empty string `""`.